File: src/java/magma/api/collect/list/ListCollector.java
package magma.api.collect.list;

import jvm.api.collect.list.Lists;
import magma.api.collect.Collector;

public class ListCollector<T> implements Collector<T, List<T>> {
    @Override
    public List<T> createInitial() {
        return Lists.empty();
    }

    @Override
    public List<T> fold(List<T> current, T element) {
        return current.addLast(element);
    }
}

File: src/node/magma/api/collect/list/ListCollector.ts
import { List } from "../../../../magma/api/collect/list/List";
import { Collector } from "../../../../magma/api/collect/Collector";
import { Lists } from "../../../../jvm/api/collect/list/Lists";
export class ListCollector<T> implements Collector<T, List<T>> {
	createInitial(): List<T> {
		return Lists.empty();
	}
	fold(current: List<T>, element: T): List<T> {
		return current.addLast(element);
	}
}
